### DISCUSSION TITLE
Templater and Calendar Plugin

### DISCUSSION THREAD

**Discussion by gallaid** [SilentVoid13/Templater#988] [Help] [answered]
Hello peeps. I am trying to find a solution to an issue I am having.

I am using Tempater and the Calendar plug-in together and need help with dates.
 I am using tp.date.now for my date getting. However, if I create a note for a future date it is an incorrect date because the tp.date.now is not the future date. Example: Today is January 10. I want to put a write down something in the daily note for January 20. When I go to create it in the Calandar plugin I click the date it creates the note but the dates are wrong because today is not the 20th the day is the 10th. Then I have to go through and change the dates. The way I use calendar and tempatar together by having the calandar plugin just put note in the folder with a title. I use "DDDD-dddd, MMMM Do YYYY" as a format but I would like to be able to set it up for my weekly and monthly. 
 
 TLDR: Is there a way to get/make a variable of the correct date the note should be as a global variable for the entire note? 
 
 I have tried to pass the note title into JS and then change the date in moment js but its not working.

**Comment by Zachatoo** [comment, marked as answer]
Checkout the reference and reference_format parameters in the [docs](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/date-module.html#tpdatenowformat-string--yyyy-mm-dd-offset-numberstring-reference-string-reference_format-string), as well as some of the examples.

```js
<% tp.date.now("DDDD-dddd, MMMM Do YYYY", 0, tp.file.title, "DDDD-dddd, MMMM Do YYYY") %>
```

To break it down:

```js
<%
tp.date.now(
"DDDD-dddd, MMMM Do YYYY", // date format you want to output
0, // offset, should be 0 if you want today, 1 for tomorrow, -1 for yesterday, etc
tp.file.title, // date referential, basically determines "what is today", in our case, it's the current note's title
"DDDD-dddd, MMMM Do YYYY" // date reference format, should match the format of the date referential
) %>
```

**Reply by gallaid** [reply]
Not sure how I didnt see that in the doc! Thanks!
